packages/core/src/elements/BasicElement.ts

```diff
@@ -175,6 +180,10 @@ export abstract class BasicElement extends LitElement {
       this.tabIndex = this.defaultTabIndex;
     }
 
+    if (!this.hasAttribute('role') && typeof this.defaultRole === 'string') {
```

Comment:
`typeof` is up to 50% faster than `hasAttribute` as `typeof` does not need to hit the DOM tree. Can you please switch the order here and for tabindex as well

Reply:
Thanks. It's done.